NuRadioReco/modules/efieldRadioInterferometricReconstruction.py

```diff
@@ -39,550 +56,834 @@ class efieldInterferometricDepthReco:
+        signal_kind: {'power', 'hilbert_sum', 'amplitude'}
+            Quantity to time-integrate. Using 'power' (the default) squares the summed trace,
+            which is recommended (square of sum if larger than sum of squares)
+        relative_signal_treshold: float, default: 0.
+            Fraction of the strongest measured fluence that should be contained
+            in a trace for it to be used in the reconstruction.
+        depths: np.ndarray, default: np.arange(400, 800, 10) * units.g / units.cm2
+            Atmospheric depths along which to perform a first coarse sampling.
+        mc_jitter: float, default: None
+            Standard deviation of Gaussian noise added to timings, in nanoseconds (internal NRR units).
+            None implies no jitter.
+        """
+        self._signal_kind = signal_kind
+        self._signal_threshold = relative_signal_treshold
+
+        self._depths = depths / units.g * units.cm2
+        self._binsize = self._depths[1] - self._depths[0]
+        self._mc_jitter = mc_jitter / units.ns
 
-        shower : BaseShower
+        self.set_station_data(evt, det, station_ids=station_ids)
+
+        if not self._debug:
+            xrit = self.reconstruct_interferometric_depth()
+        else:
+            depths_final, signals_tmp, signals_final, rit_parameters = \
+                self.reconstruct_interferometric_depth(return_profile=True)
+            xrit = rit_parameters[1]
+            ax = plt.figure().add_subplot()
+            ax.scatter(self._depths, signals_tmp, color="blue",
+                       label="Coarse sampling", s=2, zorder=1.1)
+            ax.scatter(depths_final, signals_final, color="red",
+                       label="Fine sampling", s=2, zorder=1)
+            ax.plot(depths_final, normal(depths_final, *rit_parameters),
+                    label="Gaussian fit", color="black", ls="--")
+            ax.axvline(rit_parameters[1])
+            ax.set_xlabel(r"$X [\mathrm{g}/\mathrm{cm}^2]$")
+            ax.set_ylabel(self._signal_kind)
+            ax.legend()
+            plt.show()
+
+        self._shower.set_parameter(
+            shp.interferometric_shower_maximum,
+            xrit * units.g / units.cm2)
+
+    def end(self):
         """
+        End the module
 
+        Returns
+        -------
+        self._data: dict
+            Dictionary containing some additional information which can be useful for studies.
+        """
+        return self._data
+
+    def update_atmospheric_model_and_refractivity_table(self):
+        """
+        Updates model of the atmosphere and tabulated, integrated refractive index according to shower properties.
+        Default atmospheric model is 17, if it is not found in `self._shower`.
+        """
+
+        logger.warning(
+            "flat earth geometry assumed. default was curved. If issue has been fixed, consider moving back to curved")
+        curved = False
```

Comment:
@twybo can you elaborate what issue you encountered with a curved atmosphere? In principle we can make it dependent on the shower zenith.